SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/StorageAdapter.cs

```diff
@@ -1,158 +1,193 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using Microsoft.AppCenter.Utils;
-using Microsoft.AppCenter.Utils.Files;
-using SQLite;
 using System;
 using System.Collections.Generic;
-using System.Linq.Expressions;
-using System.Reflection;
-using System.Threading.Tasks;
+using System.Linq;
+using SQLitePCL;
 
 namespace Microsoft.AppCenter.Storage
 {
     internal class StorageAdapter : IStorageAdapter
     {
-        private SQLiteAsyncConnection _dbConnection;
-        internal Directory _databaseDirectory;
-        private readonly string _databasePath;
+        private sqlite3 _db;
 
-        public StorageAdapter(string databasePath)
+        public void Initialize(string databasePath)
         {
-            _databasePath = databasePath;
-            var databaseDirectoryPath = System.IO.Path.GetDirectoryName(databasePath);
-            if (databaseDirectoryPath != string.Empty)
+            int result;
+            try
+            {
+                raw.SetProvider(new SQLite3Provider_e_sqlite3());
+                result = raw.sqlite3_open(databasePath, out _db);
+            }
+            catch (Exception e)
             {
-                _databaseDirectory = new Directory(databaseDirectoryPath);
+                throw new StorageException("Failed to open database connection.", e);
+            }
+            if (result != raw.SQLITE_OK)
+            {
+                var errorMessage = raw.sqlite3_errmsg(_db);
+                throw new StorageException($"Failed to open database connection, result={result}\n\t{errorMessage}");
             }
         }
 
-        public async Task CreateTableAsync<T>() where T : new()
+        public void Dispose()
```

Comment:
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#dispose-and-disposeboolean

The C# Dispose pattern should implement 2 dispose methods:
(1) `public void Dispose()` which can be called by the consumer of the class to dispose resources.
(2) `protected virtual void Dispose(bool disposing)` which will be called by the both the consumer and the finalizer in gc.
This implementation doesn't handle the finalizer scenario.

Reply:
@Jamminroot 

> The Dispose method performs all object cleanup, so the garbage collector no longer needs to call the objects' Object.Finalize override. Therefore, the call to the SuppressFinalize method prevents the garbage collector from running the finalizer. 

Per C# docs, the best practice is to call `GC.SuppressFinalize(this)` in `public void Dispose()` to tell GC that no need to clean resources again.

```c#
public void Dispose()
{
    Dispose(true);
    GC.SuppressFinalize(this);
}
```

Reply:
`SuppressFinalize` should only be called by a class that has a finalizer, no need for it in this case

Reply:
@Jamminroot 
Ignore my previous comment.
`GC.SuppressFinalize` is only when finalizer is implemented in the class. Since `StorageAdapater` class doesn't implement a finalizer, `GC.SuppressFinalize` is not needed.